xdsl/transforms/lower_mpi.py

```diff
@@ -742,7 +743,11 @@ def walker(op: Operation):
                                       len(module.body.blocks[0].ops))
 
 
-def lower_mpi(ctx: MLContext, module: builtin.ModuleOp):
+@dataclass
+class LowerMPIPass(OperationPass):
+
+    name = 'lower-mpi'
+
     # TODO: how to get the lib info in here?
     lib_info = MpiLibraryInfo()
```

Comment:
Not sure about this, but don't we run into the classic Python shared argument problem here? If you instantiate two of these and mutate this on one, is it visible from the other one?

Reply:
It is, yes. Not sure if that's something that we should be concerned about though : it is then registered under the initial name.

Reply:
I was talking about the lib_info, which seems more dangerous to me.

Reply:
Oh! Maybe this should just be in the function then, yes.
@AntonLydike, do you have an input on that?